### PR TITLE
`custom-property-pattern` rule changed for compatibility with WP naming

### LINKED ISSUE #2538: [bug] bud-sass/stylelint-config in incompatible with WP naming
### Agreement

- [X] This is not a duplicate of an existing issue
- [X] I have read the [guidelines for Contributing to Roots Projects](https://github.com/roots/.github/blob/master/CONTRIBUTING.md)
- [X] This is not a personal support request that should be posted on the [Roots Discourse](https://discourse.roots.io/) community

### Describe the issue

Connecting a `@roots/bud-sass/stylelint-config` causes the stylelint to identify the WP custom properties as having incorrect names 

```
Expected custom property name "--wp-admin--admin-bar--height" to be kebab-case  custom-property-pattern
```

### Expected Behavior

 `@roots/bud-sass/stylelint-config`  contains configuration changes to take into account the WP naming

### Actual Behavior

Stylelint throw error on each WP-defined custom property.

### Steps To Reproduce

Use `--wp-admin--admin-bar--height` in scss file and lint it

### version

6.20.1

### Logs

_No response_

### Configuration

_No response_

### Relevant .budfiles

_No response_

### PR DESCRIPTION
**Pull request by cawa-93**
## Type of change

**MAJOR: breaking change**
fix #2538

### CODE REVIEW

File: sources/@roots/bud-sass/config/stylelint.cjs [hunk 1]
```diff
@@ -3,4 +3,12 @@ module.exports = {
     `@roots/bud-stylelint/config`,
     `stylelint-config-recommended-scss`,
   ],
+  rules: {
+    'custom-property-pattern': [
+      '^([a-z][a-z0-9]*)(-{1,2}[a-z0-9]+)*$',
```
Comment (kellymears) [inline sources/@roots/bud-sass/config/stylelint.cjs]: What is the source for this regular expression?

It seems like all WordPress custom properties follow a form that allows this to be narrowed down a lot more.

All the documentation and devtools searching I've done has led me to believe that wp custom properties will all begin with either `wp--preset` or `wp--custom`.

Just an initial attempt to maybe improve this a little bit:

```suggestion
      '^wp--preset|wp--custom(-{1,2}[a-z0-9]+)*$',
```

But, I supsect we could also incorporate the list of possible properties as well:

https://github.com/roots/bud/blob/c91d22aa73a49412895152a7f0cc2b6a1732ec9d/sources/%40roots/wordpress-theme-json-webpack-plugin/src/theme.ts#L127-L138

Which would be something like `/wp--(preset|custom)--(blocks|border|color|spacing|etc)(-{1,2}[a-z0-9]+)*$/`

That doesn't seem to be like too much extra and would provide an appreciable difference in UX. But, I'm not accounting for any edge cases should they exist, and my regular expressions are untested (I just composed them in the github code review textarea).

But, worth considering and maybe trying to tackle.

**Comment by kellymears** [comment]
We will want to make this change here, because it is a WordPress specific rule and should not be applied to everyone using sass with bud.js.

https://github.com/roots/bud/tree/main/sources/%40roots/bud-preset-wordpress/config/stylelint.cjs#L1-L8

File: sources/@roots/bud-sass/config/stylelint.cjs (hunk 1 above)
Comment (cawa-93) [reply sources/@roots/bud-sass/config/stylelint.cjs]: > What is the source for this regular expression?

It's [`stylelint-config-standard`](https://github.com/stylelint/stylelint-config-standard/blob/f399e875cb46099bcac8aaed165a88d95736d328/index.js#L49-L54). Standard rule doesn't allow double dash `--` what WP is actually using. So I simply extend this rule to allow it.

Comment (kellymears) [reply sources/@roots/bud-sass/config/stylelint.cjs]: that makes sense. i'd be cool with approving this PR as is so as to not overcomplicate things once the code change is moved over to the proper file.

**Comment by cawa-93** [comment]
@kellymears moved

**Comment by kellymears** [comment]
Okay, I ran some tests on this and it's looking good.

Can you add the following test to `@sources/@roots/bud-preset-wordpress/test/stylelint-config.test.ts`?

```ts
import * as stylelint from 'stylelint'
import {describe, expect, it} from 'vitest'

describe(`@roots/bud-preset-wordpress/stylelint`, () => {
  it(`should error without rule`, async () => {
    const result = await stylelint.lint({
      code: `
        .foo {
          color: var(--wp--preset--color--primary);
        }
      `,
      config: {
        extends: [
          `@roots/bud-sass/stylelint-config`,
        ],
      },
    })

    expect(result.errored).toBe(true)
  })

  it(`should not error with rule`, async () => {
    const result = await stylelint.lint({
      code: `
        .foo {
          color: var(--wp--preset--color--primary);
        }
      `,
      config: {
        extends: [
          `@roots/bud-sass/stylelint-config`,
          `@roots/bud-preset-wordpress/stylelint`,
        ],
      },
    })

    expect(result.errored).toBe(false)
  })
})
```

File: sources/@roots/bud-preset-wordpress/config/stylelint.cjs [hunk 2]
```diff
@@ -3,6 +3,11 @@
  */
 module.exports = {
   rules: {
-    'custom-property-pattern': null,
+    'custom-property-pattern': [
+      '^([a-z][a-z0-9]*)(-{1,2}[a-z0-9]+)*$',
```
Comment (kellymears) [inline sources/@roots/bud-preset-wordpress/config/stylelint.cjs]: this is passing eslint cos it's a cjs file but can you convert this value to a string literal? thanks!

Comment (cawa-93) [reply sources/@roots/bud-preset-wordpress/config/stylelint.cjs]: I'm sorry, I don't understand your comment. Isn't this value already a string literal?

Comment (kellymears) [reply sources/@roots/bud-preset-wordpress/config/stylelint.cjs]: instead of `'` quotes use `. the key is fine with single quotes.

**Comment by kellymears** [comment]
thank you very much!

**Pull request merged**